packages/@aws-cdk/aws-appmesh/lib/virtual-service.ts

```diff
@@ -211,3 +179,68 @@ export interface VirtualServiceAttributes {
    */
   readonly clientPolicy?: ClientPolicy;
 }
+
+/**
+ * Properties for a VirtualService provider
+ */
+export interface VirtualServiceProviderConfig {
+  /**
+   * Virtual Node based provider
+   *
+   * @default - none
+   */
+  readonly virtualNodeProvider?: CfnVirtualService.VirtualNodeServiceProviderProperty;
+
+  /**
+   * Virtual Router based provider
+   *
+   * @default - none
+   */
+  readonly virtualRouterProvider?: CfnVirtualService.VirtualRouterServiceProviderProperty;
+}
+
+/**
+ * Represents the properties needed to define the provider for a VirtualService
+ */
+export abstract class VirtualServiceProvider {
+  /**
+   * Returns a VirtualNode based Provider for a VirtualService
+   */
+  public static virtualNode(virtualNode: IVirtualNode): VirtualServiceProvider {
+    return new VirtualServiceProviderImpl(virtualNode, undefined);
+  }
+
+  /**
+   * Returns a VirtualRouter based Provider for a VirtualService
+   */
+  public static virtualRouter(virtualRouter: IVirtualRouter): VirtualServiceProvider {
+    return new VirtualServiceProviderImpl(undefined, virtualRouter);
+  }
+
```

Comment:
Let's add a third constructor here, `mesh(mesh: IMesh): VirtualServiceProvider`.

Reply:
What is the implication of this? that there is no provider?

Reply:
I think so. If you want, you can call the method `none` instead: `none(mes: IMesh): VirtualServiceProvider`.

Reply:
Thinking on this more, our API spec does not require you to define a provider but the resource would not provide any usability in that state. Providing guardrails here ensures that customers have configured their mesh properly.